org.eclipse.xtext.extras.tests/src-gen/org/eclipse/xtext/enumrules/services/EnumRulesTestLanguageGrammarAccess.java

```diff
@@ -95,28 +96,27 @@
 		private final Keyword cDifferentNameDifferentLiteralKeyword_2_0 = (Keyword)cDifferentNameEnumLiteralDeclaration_2.eContents().get(0);
 		
 		//enum ExistingEnum:
-		//	SameName | OverriddenLiteral="overridden" | DifferentName="DifferentLiteral";
+		//  SameName | OverriddenLiteral = "overridden" | DifferentName
```

Comment:
Interesting ... the old logic using the serializer infers the value and generates an assignment, that is not present in the grammar. Of course the new logic is not capable of that.

Reply:
For completeness: The serializer operates on the model which is augmented by the Xtext linker. The linker will create objects like the return type ref or the string literal for enumerations if the short-hand notation is used.
The transient value service is later responsible for skipping these again - but it is not perfect: The string literal `DifferentLiteral` would be optional.